skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+from itertools import product
+
+from skimage.util import invert, view_as_windows
+
+
+def rolling_ball(image, radius=50, white_background=False,
+                 chunk_size=(100, 100)):
+    """Perform background subtraction using the rolling ball method.
+
+    The rolling ball algorithm estimates background intensity for a grayscale
+    image in case of uneven exposure. It is frequently used in biomedical
+    image processing and was first proposed by Stanley R. Sternberg (1983) in
+    the paper Biomedical Image Processing [1]_.
+
+    Parameters
+    ----------
+    image : ndarray
+        The image to be filtered.
+    radius : float or tuple, optional
+        The radius of the ball that is used as the structuring element. If
+        ``radius`` is a tuple it must be of the form
+        ``(space_radius, intensity_radius)``, where `space_radius` indicates
+        the radius of the ball along the spacial axis of the image and
+        ``intensity_radius`` indicates the radius along the intensity axis.
+        This is equivalent to rolling a a spheroid in the image (as opposed to
+        a ball).
+    white_background : bool, optional
+        If true, the algorithm separates dark features from a bright
+        background.
+    chunk_size : tuple, optional
+        To reduce memory usage the image is processed in chunks of
+        `chunk_size`. Larger values will make processing faster, but will also
+        consume more memory.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The image with background removed.
+    background : ndarray
+        The background that was removed.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ball
+
+    Subtract a black background from an example image
+
+    >>> image = data.coins()
+    >>> result, bg = rolling_ball(image, radius=100)
+
+    Subtract a white background from an example image
+
+    >>> image = data.page()
+    >>> result, bg = rolling_ball(image, radius=100, white_background=True)
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+    """
+
+    try:
+        if len(radius) == 2:
+            spacial_radius = radius[0]
+            intensity_vertex = radius[1]
```

Comment:
Tuple unpacking might be clearer here: `spacial_radius, intensity_vertex = radius`; you'd have to switch `IndexError` for `ValueError` in that case as well.

Reply:
That's exactly why I don't really like parameters that can plays different roles. We can get rid of all this stuff and manipulations if we directly expose the parameters `space_radius` and `intensity_radius`.

Reply:
Also also I don't understand why your tests don't fail if `radius != 2`, e.g. `radius = 1`? The `except TypeError`-block is never reached and `spacial_radius`, `intensity_vertex` are never set / defined... am I missing something here?

Reply:
If radius has no `__len__`, then `len(radius)` will raise a `TypeError`. If it doesn't support `__getitem__` it will raise `IndexError`. It works because scalars have no `__len__`. There is no doubt room for improvement.

Reply:
Ah, silly me. So `len(radius) > 2` is still a case that's not covered.

Reply:
The case is covered after refactoring 2583a3af99d33967179ce9e699b43a3c071e58d1